test/test_torch.py

```diff
@@ -5260,6 +5260,21 @@ def test_multiplication_numpy_scalar(self):
                 self.assertTrue(r2.dtype == t_dtype)
                 self.assertTrue(r2.requires_grad)
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_parse_numpy_int(self):
+        # https://github.com/pytorch/pytorch/issues/29252
+        np_val = np.ones(1, dtype=np.int64)[0]
+        self.assertEqual(type(np_val), np.int64)
+        th_val = torch.ones(size=(1,), dtype=torch.long)
+        self.assertEqual((np_val + th_val).dtype, torch.long)
+        self.assertEqual((th_val + np_val).dtype, torch.long)
+
+        # validate use of np.int in some other cases that accept python int
```

Comment:
np.int is an alias for the python built-in int type. We should call this `np.int64`